packages/kbn-unified-field-list/src/hooks/use_new_fields.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useMemo } from 'react';
+import type { DataView, DataViewField } from '@kbn/data-views-plugin/common';
+import { ExistingFieldsReader, type FieldListItem, GroupedFieldsParams } from '../..';
+
+/**
+ * This hook is used to get the new fields of previous fields for wildcards request, and merges those
+ * with the existing fields.
+ * @param dataView
+ * @param allFields
+ * @param getNewFieldsBySpec
+ * @param fieldsExistenceReader
+ */
+export function useNewFields<T extends FieldListItem = DataViewField>({
+  dataView,
+  allFields,
+  getNewFieldsBySpec,
+  fieldsExistenceReader,
+}: {
+  dataView?: DataView | null;
```

Comment:
If there is no `dataView` then it's for text-based languages and we can simply return `allFields` unmodified.

Reply:
makes, sense, done 👍